tests/wsiToDcm.cpp

```diff
@@ -136,3 +136,252 @@ TEST(compressionString, unknown) {
+  std::unique_ptr<wsiToDicomConverter::SlideLevelDim> slide_dim;
+  slide_dim = std::move(converter.getSlideLevelDim(1, NULL, NULL, false));
+  ASSERT_EQ(slide_dim->levelWidthDownsampled, 1110);
+  ASSERT_EQ(slide_dim->levelHeightDownsampled, 1483);
+  ASSERT_EQ(slide_dim->levelToGet, 0);
+  ASSERT_EQ(slide_dim->downsample, 2);
+  ASSERT_EQ(slide_dim->level, 1);
+  ASSERT_EQ(slide_dim->multiplicator, 1);
+  ASSERT_EQ(slide_dim->downsampleOfLevel, 2);
+  ASSERT_EQ(slide_dim->levelWidth, 2220);
+  ASSERT_EQ(slide_dim->levelHeight, 2967);
+  ASSERT_EQ(slide_dim->levelFrameWidth, 100);
+  ASSERT_EQ(slide_dim->levelFrameHeight, 100);
+  ASSERT_EQ(slide_dim->frameWidthDownsampled, 200);
+  ASSERT_EQ(slide_dim->frameHeightDownsampled, 200);
+  ASSERT_EQ(slide_dim->cropSourceLevelWidth, 0);
+  ASSERT_EQ(slide_dim->cropSourceLevelHeight, 0);
+  ASSERT_TRUE(slide_dim->readOpenslide);
+}
+
+TEST(getSlideLevelDim, croping_no_progressive) {
+  std::vector<int>  downsamples;
+  wsiToDicomConverter::WsiRequest request;
+  request.inputFile = tiffFileName;
+  request.outputFileMask = testPath;
+  request.frameSizeX = 100;
+  request.frameSizeY = 100;
+  request.startOnLevel = 0;
+  request.compression = dcmCompressionFromString("jpeg");
+  request.quality = 80;
+  request.imageName = "image";
+  request.studyId = "study";
+  request.seriesId = "series";
+  request.batchLimit = 100;
+  request.retileLevels = 100;
+  request.stopDownsamplingAtSingleFrame = true;
+  request.downsamples = std::move(downsamples);
+  request.debug = true;
+  wsiToDicomConverter::WsiToDcm converter(&request);
+  std::vector<int32_t> levels;
+  std::unique_ptr<wsiToDicomConverter::SlideLevelDim> smallestSlideDim;
+  std::unique_ptr<wsiToDicomConverter::SlideLevelDim> slide_dim;
+  smallestSlideDim = std::move(converter.getSmallestSlideDim(&levels));
+  slide_dim = std::move(converter.getSlideLevelDim(1,
+                                                   NULL,
+                                                   smallestSlideDim.get(),
+                                                   false));
+  ASSERT_EQ(slide_dim->levelWidthDownsampled, 1104);
+  ASSERT_EQ(slide_dim->levelHeightDownsampled, 1472);
+  ASSERT_EQ(slide_dim->levelToGet, 0);
+  ASSERT_EQ(slide_dim->downsample, 2);
+  ASSERT_EQ(slide_dim->level, 1);
+  ASSERT_EQ(slide_dim->multiplicator, 1);
+  ASSERT_EQ(slide_dim->downsampleOfLevel, 2);
+  ASSERT_EQ(slide_dim->levelWidth, 2220);
+  ASSERT_EQ(slide_dim->levelHeight, 2967);
+  ASSERT_EQ(slide_dim->levelFrameWidth, 100);
+  ASSERT_EQ(slide_dim->levelFrameHeight, 100);
+  ASSERT_EQ(slide_dim->frameWidthDownsampled, 200);
+  ASSERT_EQ(slide_dim->frameHeightDownsampled, 200);
+  ASSERT_EQ(slide_dim->frameWidthDownsampled, 200);
+  ASSERT_EQ(slide_dim->frameHeightDownsampled, 200);
+  ASSERT_EQ(slide_dim->cropSourceLevelWidth, 2220 - 2208);
+  ASSERT_EQ(slide_dim->cropSourceLevelHeight, 2967 - 2944);
+  ASSERT_TRUE(slide_dim->readOpenslide);
```

Comment:
nit: could you add a comment explaining where 2208 and 2944 come from? e.g. 2208 = lowest dim (69) * largest downsample (32)

Reply:
Added.